File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.js
// 연속적으로 나타나는 숫자는 하나만 남겨라
// 남은 수 반환할 때는 arr의 원소들의 순서는 유지해야함
function solution(arr)
{
    const result = []
    for(let i = 0; i < arr.length; i++){
        if(result[result.length-1] !== arr[i]){
            result.push(arr[i])
        }
    }
    return result
//     var answer = [];
//     let temp;
//     for(let i = 0;i < arr.length-1; i++){
//         if(arr[i] === arr[i+1]){
//             temp = arr[i+1]
//         }
//         else if(arr[i] !== arr[i+1]){
//             answer.push(temp)
//             temp = arr[i+1]
//             }
//         }
}

File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    # 연속적으로 숫자가 나오면 담지 않는다.
    # 혹은 제거한다.
    answer = []
    for index in range(len(arr)-1):
        # print(arr[index]) 1 1 3 3 0 1 1
        if arr[index] == arr[index+1]:
            continue
        else:
            answer.append(arr[index])
    answer.append(arr[len(arr)-1])
    return answer